code/modules/stock_market/computer.dm

```diff
@@ -190,30 +230,25 @@ a.updated {
 	var/avail = S.available_shares
 	var/price = S.current_value
 	var/canbuy = round(b / price)
-	var/amt = round(input(user, "How many shares? (Available: [avail], unit price: [price], can buy: [canbuy])", "Buy shares in [S.name]", 0) as num|null)
-	if (!user)
-		return
-	if (!amt)
-		return
-	if (!(user in range(1, src)))
+	var/amt = round(input(user, "How many shares? \n(Available: [avail], unit price: [price], can buy: [canbuy])", "Buy shares in [S.name]", 0) as num|null)
+	if (!user || !(user in range(1, src)))
 		return
 	if (li != logged_in)
 		return
 	b = balance()
 	if (!isnum(b))
 		user << "<span class='danger'>No active account on the console!</span>"
 		return
-	if (amt > S.available_shares)
-		user << "<span class='danger'>That many shares are not available!</span>"
-		return
-	var/total = amt * S.current_value
-	if (total > b)
-		user << "<span class='danger'>Insufficient points.</span>"
```

Comment:
Why is this check no longer in? Wouldn't you be able make transactions without the required credits?